db/migrate/20210511170803_create_invalid_contacts.rb

```diff
@@ -0,0 +1,15 @@
+class CreateInvalidContacts < ActiveRecord::Migration[6.1]
```

Comment:
Having a separate table for invalid contacts is a pretty good idea. What I'm not so sure about is if it's a good idea to save separate parsed columns for it, I think you probably want to save only the raw unparsed line.

I don't have any strong opinions about it, so it's really just a suggestion. But I have a gut feeling that by the time you want to store a `InvalidContact` record you'd already have an error going on, so it's better to not  pile more potentially complex code on top of the situation.